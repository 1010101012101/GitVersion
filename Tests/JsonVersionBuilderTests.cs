using System;
using ApprovalTests;
using GitFlowVersion;
using NUnit.Framework;

[TestFixture]
public class JsonVersionBuilderTests
{
    [Test]
    public void Json()
    {
        var semanticVersion = new VersionAndBranch
                              {
                                  BranchType = BranchType.Feature,
                                  BranchName = "feature1",
                                  Sha = "a682956dc1a2752aa24597a0f5cd939f93614509",
                                  Version = new SemanticVersion
                                                       {
                                                           Major = 1,
                                                           Minor = 2,
                                                           Patch = 3,
                                                           Stability = Stability.Unstable,
                                                           PreReleasePartOne = 4,
                                                           Suffix = "a682956d",
                                                       }
                              };
        var json = semanticVersion.ToJson().Replace("\r\n", Environment.NewLine);
        Approvals.Verify(json);
    }

}