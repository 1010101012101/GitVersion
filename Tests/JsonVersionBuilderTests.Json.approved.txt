{
  "Major":1,
  "Minor":2,
  "Patch":3,
  "PreReleaseTag":"unstable.4",
  "PreReleaseTagWithDash":"-unstable.4",
  "Suffix":"a682956d",
  "InformationalVersion":"1.2.3-unstable.4+a682956d.Branch.feature1.Sha.a682956dc1a2752aa24597a0f5cd939f93614509",
  "MajorMinorPatch":"1.2.3",
  "SemVer":"1.2.3-unstable.4",
  "SemVerPadded":"1.2.3-unstable.0004",
  "AssemblySemVer":"1.2.3.0",
  "FullSemVer":"1.2.3-unstable.4",
  "FullSemVerPadded":"1.2.3-unstable.0004+a682956d",
  "ClassicVersion":"1.2.3.4",
  "BranchName":"feature1",
  "BranchType":"Feature",
  "Sha":"a682956dc1a2752aa24597a0f5cd939f93614509"
}